dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -465,6 +488,42 @@ private boolean isServerError(Exception ex) {
         return false;
     }
 
+    private boolean shouldProcessPush(GitHubSourceCodeRepo gitHubSourceCodeRepo, String repository, String reference, String afterCommit) {
+        // If there's no "after" commit, process the push.
+        if (afterCommit == null) {
+            return true;
+        }
+        try {
+            // Retrieve the "current" head commit.
+            String currentCommit = getHeadCommit(gitHubSourceCodeRepo, repository, reference);
+            LOG.info("afterCommit={} currentCommit={}", afterCommit, currentCommit);
+            // Process the push iff the "current" and "after" commit hashes are equal.
+            // If the repo's "current" hash doesn't match the event's "after" hash, the repo has changed since the event was created.
+            // The "after" commit hash cannot be null here, so we'll return false (ignore the push) if the "current" commit hash is null (because the ref no longer exists).
+            // Later, another event corresponding to the subsequent change will arrive and trigger an update.
+            return Objects.equals(afterCommit, currentCommit);
+        } catch (CustomWebApplicationException ex) {
+            // If there's a problem determining if the push needs processing, assume it does.
+            LOG.info("CustomWebApplicationException determining whether to process push", ex);
+            return true;
+        }
+    }
+
+    private boolean shouldProcessDelete(GitHubSourceCodeRepo gitHubSourceCodeRepo, String repository, String reference) {
+        try {
+            // Process the delete if the reference does not exist (there is no head commit).
+            return getHeadCommit(gitHubSourceCodeRepo, repository, reference) == null;
+        } catch (CustomWebApplicationException ex) {
+            // If there's a problem determining if the delete needs processing, assume it does.
+            LOG.info("CustomWebApplicationException determining whether to process delete", ex);
+            return true;
+        }
+    }
+
+    private String getHeadCommit(GitHubSourceCodeRepo repo, String repository, String ref) {
+        return repo.getCommitID(repository, ref, false);
```

Comment:
It looks like this function is used to determine if the `after` commit is the head commit. Perhaps this endpoint might be better suited and it'll save us some API calls: 
- https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-branches-for-head-commit
- https://github-api.kohsuke.org/apidocs/org/kohsuke/github/GHCommit.html#listBranchesWhereHead()

You can call this endpoint with the `after` commit and if it is the head commit, then it will return the branches where it's the head commit.

Reply:
Thanks for pointing out the endpoint, but we'd still need to handle tags, for which there's no analogous kohsuke call (that I can find, and possibly no corresponding github endpoint).  So, we could code something that might save 0.6 or so API calls per push, on average, but imho, it's not worth the code complexity.